conference.js

```diff
@@ -897,41 +724,28 @@ export default {
             return this._setLocalAudioVideoStreams(tracks);
         }
 
-        const [ tracks, con ] = await this.createInitialLocalTracksAndConnect(roomName, initialOptions);
-
-        this._initDeviceList(true);
+        const { tryCreateLocalTracks, errors } = this.createInitialLocalTracks(initialOptions);
 
-        const filteredTracks = handleInitialTracks(initialOptions, tracks);
+        return Promise.all([
+            tryCreateLocalTracks.then(tr => {
+                this._displayErrorsForCreateInitialLocalTracks(errors);
 
-        setGUMPendingStateOnFailedTracks(filteredTracks);
+                return tr;
+            }).then(tr => {
+                this._initDeviceList(true);
 
-        return this.startConference(con, filteredTracks);
-    },
+                const filteredTracks = handleInitialTracks(initialOptions, tr);
 
-    /**
-     * Joins conference after the tracks have been configured in the prejoin screen.
-     *
-     * @param {Object[]} tracks - An array with the configured tracks
-     * @returns {void}
-     */
-    async prejoinStart(tracks) {
-        if (!_connectionPromise) {
-            // The conference object isn't initialized yet. Wait for the promise to initialise.
-            await new Promise(resolve => {
-                _onConnectionPromiseCreated = resolve;
-            });
-            _onConnectionPromiseCreated = undefined;
-        }
+                setGUMPendingStateOnFailedTracks(filteredTracks);
 
-        let con;
+                this._setLocalAudioVideoStreams(filteredTracks);
```

Comment:
I don't think this is needed anymore since we are anyway executing it as part of `startConference` in `_createRoom`.

Reply:
But this does the addLocalTrack/replaceLocalTrack to the store. We want to keep it to be earlier ... aa this is the non prejoin case ... and it will be set in startConference in the promise after the track creation and connection. 
Ooo and I need to return the filtered tracks to be used for startConference